test/HttpApiClientTest.ts

```diff
@@ -0,0 +1,116 @@
+import {HttpApiClient, HttpClient} from "../src/httpClient";
+
+describe("HttpApiClient", () => {
+    let httpApiClient: HttpApiClient;
+
+    it("should return a valid baseUrl after construction", () => {
```

Comment:
Split this test into 2 separate tests as both scenarios are quite different and the title actually may not fit the first case. How about *it should leave the baseUrl untouched if omitted during construction*